NuRadioMC/simulation/simulation.py

```diff
@@ -147,112 +1095,134 @@ def __init__(self, inputfilename,
+            logger.error("Please provide both detector_simulation_filter_amp and detector_simulation_trigger or detector_simulation_part1 and detector_simulation_part2")
+            raise ValueError("Please provide both detector_simulation_filter_amp and detector_simulation_trigger or detector_simulation_part1 and detector_simulation_part2")
+        if((self.detector_simulation_part1 is not None) and (self.detector_simulation_filter_amp is not None)):
+            logger.error("Please provide either detector_simulation_filter_amp and detector_simulation_trigger or detector_simulation_part1 and detector_simulation_part2")
+            raise ValueError("Please provide either detector_simulation_filter_amp and detector_simulation_trigger or detector_simulation_part1 and detector_simulation_part2")
 
         # Initialize detector
+        self._antenna_pattern_provider = antennapattern.AntennaPatternProvider()
         if det is None:
-            logger.status("Detectorfile {}".format(os.path.abspath(self._detectorfile)))
-            kwargs = dict(json_filename=self._detectorfile, default_station=default_detector_station,
+            logger.status("Detectorfile {}".format(os.path.abspath(detectorfile)))
+            kwargs = dict(json_filename=detectorfile, default_station=default_detector_station,
                               default_channel=default_detector_channel, antenna_by_depth=False)
             kwargs.update(det_kwargs)
             self._det = detector.Detector(**kwargs)
         else:
             self._det = det
 
-        self._det.update(evt_time)
+        self._det.update(self._evt_time)
+
+        # initialize propagation module
+        if self._config['propagation']['ice_model'] == "custom":
+            if ice_model is None:
+                logger.error("ice model is set to 'custom' in config file but no custom ice model is provided.")
+                raise AttributeError("ice model is set to 'custom' in config file but no custom ice model is provided.")
+            self._ice = ice_model
+        else:
+            self._ice = NuRadioMC.utilities.medium.get_ice_model(self._config['propagation']['ice_model'])
+        prop = propagation.get_propagation_module(self._config['propagation']['module'])
+        self._propagator = prop(
+            self._ice, self._config['propagation']['attenuation_model'],
+            log_level=self._log_level_ray_propagation,
+            n_frequencies_integration=int(self._config['propagation']['n_freq']),
+            n_reflections=int(self._config['propagation']['n_reflections']),
+            config=self._config,
+            detector=self._det
+        )
 
         self._station_ids = self._det.get_station_ids()
         self._event_ids_counter = {}
         for station_id in self._station_ids:
             self._event_ids_counter[station_id] = -1  # we initialize with -1 becaue we increment the counter before we use it the first time
```

Comment:
`self._event_ids_counter = {station_id: -1 for station_id in self._station_ids}`